src/main/java/org/icgc/argo/program_service/services/EgoService.java

```diff
@@ -390,10 +401,116 @@ public static GroupName createProgramGroupName(String name, UserRole role){
   }
 
 
+  public void updateUserRole(@NonNull UUID userId, @NonNull UUID programId, @NonNull UserRole role) throws NotFoundException, RuntimeException {
+    getUserById(userId);
+    val program = findProgramById(programId);
+    val shortname = program.getShortName();
+
+    val groups = getGroupsFromUser(userId);
+    if(groups.isEmpty()){
+      throw new NotFoundException(String.format("No groups found for user id %s.", userId));
+    }
+    // determine the current group
+    for(val group : groups){
+      if(group.getName().toLowerCase().contains(shortname.toLowerCase())) {
+        if(!isSameRole(role, group.getName())){
+          removeUserFromCurrentGroup(group, userId);
+        } else {
+          throw new RuntimeException(format("Cannot update user role %s, new role is the same as current role.", role.toString()));
+        }
+      }
+    }
+    val programEgoGroup = programEgoGroupRepository.findByProgramIdAndRole(programId, role);
+    val egoGroupId = programEgoGroup.get().getEgoGroupId();
+    addUserToGroup(userId, egoGroupId);
+  }
+
+  private boolean isSameRole(@NonNull UserRole role, @NonNull String groupName) throws RuntimeException {
+    UserRole currentRole = UserRole.UNRECOGNIZED;
+    if (groupName.contains(UserRole.COLLABORATOR.toString())) {
+      currentRole = UserRole.COLLABORATOR;
+    } else if (groupName.contains(UserRole.SUBMITTER.toString())) {
+      currentRole = UserRole.SUBMITTER;
+    } else if (groupName.contains(UserRole.CURATOR.toString())) {
+      currentRole = UserRole.CURATOR;
+    } else if (groupName.contains(ADMIN.toString())) {
+      currentRole = ADMIN;
+    } else if (groupName.contains(UserRole.BANNED.toString())) {
+      currentRole = UserRole.BANNED;
+    } else {
+      log.error("Unrecognized role {}.", currentRole.toString());
+      throw new RuntimeException("Unrecognized role!");
```

Comment:
Try to be as descriptive as possible. RuntimeException is very generic. Something like `IllegalArgumentException` would be more appropriate